### PR TITLE
Introduce `process.*` field set reuse filtering

### PR DESCRIPTION
**Pull request by ebeahan**
Follow-up to #1842

The [Linux event model](https://github.com/elastic/ecs/blob/main/rfcs/text/0030-linux-event-model.md) requires many reuses of the `process.*` field set, but not all these reuses need the entire field set be reused.

This change leverages the `--subset` feature to limit the number of new fields introduced by these complex reuses. No existing `process.*` or `process.parent.*` fields are affected or removed by this change.

### CODE REVIEW

File: experimental/generated/csv/fields.csv [hunk 1]
```diff
@@ -1006,207 +607,47 @@ ECS_Version,Indexed,Field_Set,Field,Type,Level,Normalization,Example,Description
...
-8.3.0-dev+exp,true,process,process.entry_meta.source.nat.ip,ip,extended,,,Source NAT ip
-8.3.0-dev+exp,true,process,process.entry_meta.source.nat.port,long,extended,,,Source NAT port
-8.3.0-dev+exp,true,process,process.entry_meta.source.packets,long,core,,12,Packets sent from the source to the destination.
-8.3.0-dev+exp,true,process,process.entry_meta.source.port,long,core,,,Port of the source.
-8.3.0-dev+exp,true,process,process.entry_meta.source.registered_domain,keyword,extended,,example.com,"The highest registered source domain, stripped of the subdomain."
-8.3.0-dev+exp,true,process,process.entry_meta.source.subdomain,keyword,extended,,east,The subdomain of the domain.
-8.3.0-dev+exp,true,process,process.entry_meta.source.top_level_domain,keyword,extended,,co.uk,"The effective top level domain (com, org, net, co.uk)."
-8.3.0-dev+exp,true,process,process.entry_meta.type,keyword,extended,,,The entry type for the entry session leader.
-8.3.0-dev+exp,true,process,process.env_vars,object,extended,,"{""USER"": ""elastic"",""LANG"": ""en_US.UTF-8"",""HOME"": ""/home/elastic""}",Environment variables set at the time of the event.
```
Comment (kgeller) [inline experimental/generated/csv/fields.csv:1050]: Did we want to remove `process.env_vars`?

File: experimental/generated/csv/fields.csv [hunk 2]
```diff
@@ -1216,7 +657,6 @@ ECS_Version,Indexed,Field_Set,Field,Type,Level,Normalization,Example,Description
 8.3.0-dev+exp,true,process,process.hash.sha512,keyword,extended,,,SHA512 hash.
 8.3.0-dev+exp,true,process,process.hash.ssdeep,keyword,extended,,,SSDEEP hash.
 8.3.0-dev+exp,true,process,process.hash.tlsh,keyword,extended,,,TLSH hash.
-8.3.0-dev+exp,true,process,process.interactive,boolean,extended,,True,Whether the process is connected to an interactive shell.
```
Comment (kgeller) [inline experimental/generated/csv/fields.csv:1219]: I think we still want `process.interactive`

https://github.com/elastic/endpoint-package/blob/49d9a9ac02762afa3ad158febe5edc164883d715/custom_subsets/elastic_endpoint/process/linux_event_model_event.yaml#L34

File: experimental/generated/csv/fields.csv [hunk 3]
```diff
@@ -1263,203 +703,21 @@ ECS_Version,Indexed,Field_Set,Field,Type,Level,Normalization,Example,Description
...
-8.3.0-dev+exp,true,process,process.parent.entry_meta.source.nat.port,long,extended,,,Source NAT port
-8.3.0-dev+exp,true,process,process.parent.entry_meta.source.packets,long,core,,12,Packets sent from the source to the destination.
-8.3.0-dev+exp,true,process,process.parent.entry_meta.source.port,long,core,,,Port of the source.
-8.3.0-dev+exp,true,process,process.parent.entry_meta.source.registered_domain,keyword,extended,,example.com,"The highest registered source domain, stripped of the subdomain."
-8.3.0-dev+exp,true,process,process.parent.entry_meta.source.subdomain,keyword,extended,,east,The subdomain of the domain.
-8.3.0-dev+exp,true,process,process.parent.entry_meta.source.top_level_domain,keyword,extended,,co.uk,"The effective top level domain (com, org, net, co.uk)."
-8.3.0-dev+exp,true,process,process.parent.entry_meta.type,keyword,extended,,,The entry type for the entry session leader.
-8.3.0-dev+exp,true,process,process.parent.env_vars,object,extended,,"{""USER"": ""elastic"",""LANG"": ""en_US.UTF-8"",""HOME"": ""/home/elastic""}",Environment variables set at the time of the event.
```
Comment (kgeller) [inline experimental/generated/csv/fields.csv:1293]: Another ? for `process.parent.env_vars`

I don't see any `env_vars` in https://github.com/elastic/endpoint-package/blob/49d9a9ac02762afa3ad158febe5edc164883d715/custom_subsets/elastic_endpoint/process/linux_event_model_event.yaml , but I know we added it for it, so just double checking

File: experimental/generated/csv/fields.csv [hunk 4]
```diff
@@ -1263,203 +703,21 @@ ECS_Version,Indexed,Field_Set,Field,Type,Level,Normalization,Example,Description
...
-8.3.0-dev+exp,true,process,process.parent.group_leader.working_directory,keyword,extended,,/home/alice,The working directory of the process.
-8.3.0-dev+exp,true,process,process.parent.group_leader.working_directory.text,match_only_text,extended,,/home/alice,The working directory of the process.
 8.3.0-dev+exp,true,process,process.parent.hash.md5,keyword,extended,,,MD5 hash.
 8.3.0-dev+exp,true,process,process.parent.hash.sha1,keyword,extended,,,SHA1 hash.
 8.3.0-dev+exp,true,process,process.parent.hash.sha256,keyword,extended,,,SHA256 hash.
 8.3.0-dev+exp,true,process,process.parent.hash.sha384,keyword,extended,,,SHA384 hash.
 8.3.0-dev+exp,true,process,process.parent.hash.sha512,keyword,extended,,,SHA512 hash.
 8.3.0-dev+exp,true,process,process.parent.hash.ssdeep,keyword,extended,,,SSDEEP hash.
 8.3.0-dev+exp,true,process,process.parent.hash.tlsh,keyword,extended,,,TLSH hash.
-8.3.0-dev+exp,true,process,process.parent.interactive,boolean,extended,,True,Whether the process is connected to an interactive shell.
```
Comment (kgeller) [inline experimental/generated/csv/fields.csv:1462]: https://github.com/elastic/endpoint-package/blob/49d9a9ac02762afa3ad158febe5edc164883d715/custom_subsets/elastic_endpoint/process/linux_event_model_event.yaml#L82

File: experimental/generated/csv/fields.csv (hunk 1 above)
Comment (mitodrummer) [reply experimental/generated/csv/fields.csv:1050]: Good catch, I may have missed that in my custom_subset in endpoint-package.

File: experimental/generated/csv/fields.csv (hunk 3 above)
Comment (mitodrummer) [reply experimental/generated/csv/fields.csv:1293]: I would say only top level process gets env_vars. I will submit another PR to endpoint-package.

File: experimental/generated/csv/fields.csv (hunk 1 above)
Comment (ebeahan) [reply experimental/generated/csv/fields.csv:1050]: No - I resolved in https://github.com/elastic/ecs/pull/1847/commits/574862abfa4ddae44efa543a8b2b183229c68296

File: experimental/generated/csv/fields.csv (hunk 2 above)
Comment (ebeahan) [reply experimental/generated/csv/fields.csv]: Good catch - addressed in https://github.com/elastic/ecs/pull/1847/commits/fec3da59c2089dc1b093eae576fb39018278f977

File: experimental/generated/csv/fields.csv (hunk 4 above)
Comment (ebeahan) [reply experimental/generated/csv/fields.csv]: Addressed in https://github.com/elastic/ecs/pull/1847/commits/fec3da59c2089dc1b093eae576fb39018278f977

File: experimental/generated/csv/fields.csv (hunk 3 above)
Comment (ebeahan) [reply experimental/generated/csv/fields.csv:1293]: I added `process.parent.env_vars` to keep `process.*` and `process.parent.*` more symmetrical. Open to changing if there's any differing views.

**Review by kgeller — APPROVED** [review]
LGTM

**Comment by mitodrummer** [comment]
endpoint-package updated to include mapping for process.env_vars
https://github.com/elastic/endpoint-package/pull/240

**Review by mitodrummer — APPROVED** [review]
LGTM!

**Comment by ebeahan** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.2|[](https://github.com/elastic/ecs/pull/1848)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**